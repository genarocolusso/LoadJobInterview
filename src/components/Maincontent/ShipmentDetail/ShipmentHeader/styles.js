import styled from "styled-components";

import { MdArrowForward } from "react-icons/md";

export const Stop = styled.div`
display: inline-block`;

export const DestinationArrow = styled(MdArrowForward)`
  color: #b9b9b9;
  margin: 0px 4px;
  display: inline-block
`;


export const Container = styled.div`
  padding: 16px 12px;
  background-color: var(--content-box-bg);
  color: var(--darkgray);
  border: 1px solid var(--border-color);
  display: flex;
  margin-bottom: 8px;

  align-items: center;
  font-size: 21px;
  font-weight: 600;
  > div {
    display: flex; 
  align-items: center;
  }
  > span {
    color: var(--green);
    margin-right: 16px;
  }

  > * {
    display: inline-block
  }
`;