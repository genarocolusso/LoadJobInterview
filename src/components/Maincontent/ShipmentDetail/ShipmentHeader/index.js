import React from "react";
import { getDate } from "../../../util";

import {Container, Stop, DestinationArrow} from './styles' 

 

const  ShipmentHeader = ({stops}) =>{
 
  const showDate = ( ) => {
    // shows first stop date
    return (
      getDate(`${ stops[0].windowStart}`, "MMM") +
      " " +
      getDate(`${ stops[0].windowStart}`, "DD")
    );
  };
  

  return (
    <Container>
    <span>{showDate(stops).toLocaleUpperCase()}</span>

    { stops.map((stop, idx) => {
      //add destinations on header with arrow icon
      return (
        <div key={stop.uid}>
          <Stop>
            {stop.city}, {stop.state}
          </Stop>
          {idx <  stops.length - 1 ? (
            <DestinationArrow />
          ) : null}
        </div>
      );
    })}
  </Container>
  )
}

export default ShipmentHeader