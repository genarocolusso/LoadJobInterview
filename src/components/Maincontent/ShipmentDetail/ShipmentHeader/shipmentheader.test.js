import React from "react";
import renderer from "react-test-renderer";

import ShipmentHeader from ".";

describe("My Connected React-Redux Component", () => {
  let component;
  const stops = [
    {
      city: "Savannah",
      state: "GA",
      country: "USA",
      windowStart: "2021-02-26T14:00:00.000Z",
      windowEnd: "2021-02-26T22:00:00.000Z",
      uid: 0,
    },
    {
      city: "Aurora",
      state: "IL",
      country: "USA",
      windowStart: "2021-02-26T14:00:00.000Z",
      windowEnd: "2021-02-26T22:00:00.000Z",
      uid: 1,
    },
  ];

  component = renderer.create(<ShipmentHeader stops={stops} />);

  it("should render component with given stops", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  
});
