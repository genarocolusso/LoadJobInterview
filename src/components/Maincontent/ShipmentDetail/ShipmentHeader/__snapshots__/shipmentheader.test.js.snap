// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render component with given stops 1`] = `
<div
  className="sc-dlfnbm kfsYjI"
>
  <span>
    FEB 26
  </span>
  <div>
    <div
      className="sc-bdfBwQ hEePOT"
    >
      Savannah
      , 
      GA
    </div>
    <svg
      className="sc-gsTCUz hZFRFe"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
      />
    </svg>
  </div>
  <div>
    <div
      className="sc-bdfBwQ hEePOT"
    >
      Aurora
      , 
      IL
    </div>
  </div>
</div>
`;
